# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/api/FileSystemHandle.cpp
# Commit: 214c4d6bf7f9
# Full Hash: 214c4d6bf7f98a332c4ab81d325d9245ae810955
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-10-20 21:51:26
# Regressor Bug: 1792245
# File Overlap Count: 2
# Description:
#   Bug 1792245 - Add ErrorResult to FileSystemRequestHandler methods; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159572
# ==============================================================================

diff -r 330c69218a93 -r 214c4d6bf7f9 dom/fs/api/FileSystemHandle.cpp
--- a/dom/fs/api/FileSystemHandle.cpp	Thu Oct 20 07:33:41 2022 +0300
+++ b/dom/fs/api/FileSystemHandle.cpp	Thu Oct 20 04:21:30 2022 +0000
@@ -152,9 +152,14 @@
     fs::FileSystemChildMetadata newMetadata;
     newMetadata.parentId() = aParentId;
     newMetadata.childName() = aName;
-    mRequestHandler->MoveEntry(mManager, this, mMetadata, newMetadata, promise);
+    mRequestHandler->MoveEntry(mManager, this, mMetadata, newMetadata, promise,
+                               aError);
   } else {
-    mRequestHandler->RenameEntry(mManager, this, mMetadata, name, promise);
+    mRequestHandler->RenameEntry(mManager, this, mMetadata, name, promise,
+                                 aError);
+  }
+  if (aError.Failed()) {
+    return nullptr;
   }
 
   // Other handles to this will be broken, and the spec is ok with this, but we